packages/pages/src/util/runtime.test.ts

```diff
@@ -168,3 +96,45 @@ describe("runtime", () => {
     global.process = originalProcess;
   });
 });
+
+const setupMockEnvironment = (
+  simulatedEnvironment?: "node" | "deno"
+): NodeJS.Process => {
+  const originalProcess = process;
```

Comment:
Would something like the below work?

```ts
import browserOrNode from 'browser-or-node'

const setupMockEnvironment = (env: 'node' | 'deno') => {
  jest.spyOn(browserOrNode).mockImplementation(() => {
    // mock whatever you want directly based on whether env is deno or node
  })
}

Reply:
I had attempted that as you suggested in a prior comment. The error from jest states you cannot mock a boolean literal only a function that returns a boolean (or whatever else what you're mocking returns).

Reply:
Oooh gotcha. I gave it a quick try and I think something doMock should work and would simplify this a bit?
```ts

const setupMockEnvironment = (
  simulatedEnvironment?: "node" | "deno"
) => {
  jest.doMock('browser-or-node', () => ({
    isDeno: simulatedEnvironment === "deno",
    isNode: simulatedEnvironment === "node"
  }))
};
```

Reply:
I don't think this is technically any different than mocking at the beginning of the file and then setting the value of both variables in their if statements. I would like to continue with the current approach because it allows an environment to be entirely set in a single scope. This makes it easier to understand instead of splitting it into multiple environment checks.

Reply:
I feel like jest.doMock is more explicit that you're mocking something, as opposed to taking `browserOrNode`, asserting it to be `any` (which I think eslint will complain about if we add the @yext/eslint-config rules?)

Reply:
Is that snippet something you tried yourself? Trying it where you indicated I do not see the mock working @oshi97

Reply:
TIL to use doMock() you have to reimport modules after calling it